File: common/src/main/java/com/ssblur/scriptor/word/descriptor/duration/SimpleDurationDescriptor.java
package com.ssblur.scriptor.word.descriptor.duration;

import com.ssblur.scriptor.api.word.Descriptor;

public class SimpleDurationDescriptor extends Descriptor implements DurationDescriptor {
  Cost cost;
  double duration;
  boolean duplicates;

  public SimpleDurationDescriptor(int cost, double duration) {
    this.cost = new Cost(cost, COSTTYPE.ADDITIVE);
    this.duration = duration;
    this.duplicates = false;
  }

  @Override
  public Cost cost() {
    return cost;
  }

  @Override
  public double durationModifier() {
    return duration;
  }

  @Override
  public boolean allowsDuplicates() {
    return duplicates;
  }

  public SimpleDurationDescriptor allowDuplication() {
    this.duplicates = true;
    return this;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/duration/SimpleDurationDescriptor.kt
package com.ssblur.scriptor.word.descriptor.duration

import com.ssblur.scriptor.api.word.Descriptor

open class SimpleDurationDescriptor(cost: Int, var duration: Double): Descriptor(), DurationDescriptor {
  var cost: Cost = Cost(cost.toDouble(), COSTTYPE.ADDITIVE)
  var duplicates: Boolean = false

  override fun cost() = cost
  override fun durationModifier() = duration
  override fun allowsDuplicates() = duplicates

  fun allowDuplication(): SimpleDurationDescriptor {
    this.duplicates = true
    return this
  }
}
